yt_dlp/extractor/nhk.py

```diff
@@ -334,3 +335,140 @@ def _real_extract(self, url):
+
+        info = {}
+
+        url = file['file_name']
+        info['id'] = programme_id + '_' + headline['headline_id']
+        info['formats'] = self._extract_m3u8_formats(url, info['id'])
+        info['title'] = file.get('file_title')
+        info['description'] = file.get('file_title_sub')
+        # info['webpage_url'] = file.get('share_url')
+        info['thumbnail'] = headline.get('headline_image')
+        info['container'] = 'm4a_dash'  # force fixup so seeking works
+        info['was_live'] = True  # it's radio catch-up
+
+        aa_vinfo4 = file.get('aa_vinfo4')
+        # have to use whatever this is to get an actual air date
+        # there is an onair_date var, but it's natural language that doesnt have half of what we need anyway
+        info['release_timestamp'] = self._decode_time(aa_vinfo4.split('_')[0])
+        # open_time is when it was put on vod
+        info['timestamp'] = self._decode_time(file.get('open_time'))
+
+        return info
+
+    def _real_extract(self, url):
+        video_id, site_id, corner_id, headline_id = self._match_valid_url(url).groups()
+
+        programme_id = f'{site_id}_{corner_id}'
+        json_url = f'https://www.nhk.or.jp/radioondemand/json/{site_id}/bangumi_{programme_id}.json'
+        meta = self._download_json(json_url, programme_id)['main']
+
+        series = meta.get('program_name')
+        channel = meta.get('media_name')
+        thumbnail = meta.get('thumbnail_c') or meta.get('thumbnail_p')
+
+        detail_list = meta.get('detail_list')
+
+        result = None
+
+        # if it's 0123_45_6789 - a specific episode (as opposed to 0123_45 - the whole programme)
+        if headline_id is not None:
+            # we are downloading a single episode
+            headline = (next((i for i in detail_list if i.get('headline_id') == headline_id), None))
+            result = self._extract_episode_info(headline, programme_id)
+            result['series'] = series
+            result['channel'] = channel
+            result['thumbnail'] = result.get('thumbnail') or thumbnail
+        else:
+            # we are downloading a whole playlist/show
+            series_description = meta.get('site_detail')
+            # series_url = meta.get('share_url')
+            result = {
+                '_type': 'playlist',
+                'channel': channel,
+                'thumbnail': thumbnail,
+                'title': series,
+                'description': series_description,
+                # 'webpage_url': series_url,
+                'id': video_id,
+                'entries': [],
+            }
+            for headline in detail_list:
+                ep = self._extract_episode_info(headline, programme_id)
+                ep['series'] = series
+                ep['channel'] = channel
+                ep['thumbnail'] = ep.get('thumbnail') or thumbnail
+                result['entries'].append(ep)
+        return result
```

Comment:
I would simplify this with `traverse_obj` and `playlist_result`

 - `traverse_obj(meta, ('detail_list', ..., {dict}))` will branch out and return all dicts in the list
 - `traverse_obj(meta, ('detail_list', lambda _, v: v['headline_id'] == headline_id))` will filter for dicts in the list where the `'headline_id'` key's value is the value of the `headline_id` variable
 - if a `traverse_obj` path branches, it will always return a list unless `get_all=False`, then only 1 value will be returned

```suggestion
        if headline_id:
            return self._extract_episode_info(
                traverse_obj(meta, (
                    'detail_list', lambda _, v: v['headline_id'] == headline_id), get_all=False),
                programme_id, series_meta)

        def entries():
            for headline in traverse_obj(meta, ('detail_list', ..., {dict})):
                yield self._extract_episode_info(headline, programme_id, series_meta)

        return self.playlist_result(
            entries(), programme_id, playlist_description=meta.get('site_detail'), **series_meta)
```